### ISSUE TITLE
Deleted Files are stored in snapshots forever

### ISSUE THREAD

**Issue by Germar** [bit-team/backintime#7]
The classic example of this is when Back In Time backs up your desktop.

For example.

I create a file called 'foo' on my desktop to write some quick notes in.

I create a snapshot, and '~/Desktop/foo' is now stored in 'Snapshot1'.

I delete 'foo' from my Desktop, as I no longer need the file.

I create another snapshot 'Snapshot2'. 'Foo' is still listed in 'Snapshot2' as a file (although obviously it hasn't changed). 

For each snapshot that is taken after this, 'foo' will always be listed as a file in the snapshot, and never removed from the series of backups.

What should happen is, 'foo' should be stored in 'Snapshot1', but should not be stored in 'Snapshot2'.

When 'Snapshot1' is deleted (either manually or automatically), 'foo' is no longer backed up through back in time, as it no longer exists on the system.

The only current workaround I have found, is manually deleting the files I want to be removed from snapshots, so they are not included in the next sync.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2009-07-29T01:20:01Z
 * owner: mark-mandel
 * assignee: bratdaking
 * the launchpad url was https://bugs.launchpad.net/bugs/406092

**Comment by Germar** [comment]
(by mark-mandel)
I also realised something else as well with this bug.

Basically every file that you ever ended up in your Trash folder that got caught in a snapshot, is backed up for perpetuity.

This makes backup size grow in size at a remarkable rate.

The other workaround I using at the moment is to start my snapshots from scratch ever month or so, and therefore start from a clean slate.

**Comment by Germar** [comment]
(by bratdaking)
I tried to reproduce this bug, however in my case the foo file (and an additional test folder) gets deleted from the snapshots (I also checked the actual backup folders). 

I use backintime 0.9.26, with rsync 3.0.5 (that is the actual program that is should take track of any changes), Ubuntu 9.04
on an ext3 formatted disk. Do you accidentally store the backups on a FAT32 or NTFS partition?

Your backups shouldn't grow in size actually by this behaviour as unchanged files are actually linked by hardlinks (so it is copied only once and then linked by all the snapshots that have the file. Only if the last snapshot that contains the file is deleted, e.g. by the smart-remove option, the file will be deleted itself. This behaviour saves an incredibly lot of space and makes it possible to make a snapshot every 10 minutes without consuming much additional space (only changed files will be included additionally, rest will be hardlinked), only as far as I now this is not possible on FAT32 or NTFS drives, as they do not support hardlinked files, what could result in the behaviour you describe.

**Comment by Germar** [comment]
(by mark-mandel)
I'm not sure what is going on.

I just checked out my backups that were created after I cleared out the series of backups I had that were showing this issue, and it does not seem to exist any more.

I can clearly see items disappearing from the timeline view in back in time, whereas before it didn't.

It sounds like I may have to eat some humble pie and classify this as a weird anomaly or something.

I'll let you guys know if it shows up again, but I cross fingers that it doesn't.

Mark

**Comment by Germar** [comment]
(by m-s-whitaker)
I experienced exactly the same behaviour - somehow it sorted itself out after a few weeks, and it seems fine now. Might it have something to do with the pruning? It may be my imagination but the disappearance of this issue seemed to coincide with my disabling the auto-delete option... Let me know if you'd like me to provide any specific information.

**Comment by Germar** [comment]
(by bratdaking)
I will set the status to fixed released then. If it returns just change the status again or file a new bug.

Cheers,
Bart

**Comment by Germar** [comment]
(by thomas-bernhofer)
I excluded ".gvfs" in the Settings Dialog and now everything works.

**Comment by Germar** [comment]
(by petit-prince)
As a preparation to upgrade from Jaunty to Karmic, I backed up my home profile to an external EXT3 2.5" USB 2.0 hard drive (from scratch, no previous snapshots existed) => Snapshot A.
- Created a new file foo on my desktop, took a snapshot => Snapshot B (contains the new file foo).
- Deleted the newly created file foo, took another snapshot => Snapshot C. Snapshot C still contains the file foo, even though it had been deleted!

Bart, please let me know how I can further pinpoint this bug.

**Comment by Germar** [comment]
(by bratdaking)
And after the upgrade?

2009/10/29 petit-prince <email address hidden>

> As a preparation to upgrade from Jaunty to Karmic, I backed up my home
> profile to an external EXT3 2.5" USB 2.0 hard drive (from scratch, no
> previous snapshots existed) => Snapshot A.
> - Created a new file foo on my desktop, took a snapshot => Snapshot B
> (contains the new file foo).
> - Deleted the newly created file foo, took another snapshot => Snapshot C.
> Snapshot C still contains the file foo, even though it had been deleted!
>
> Bart, please let me know how I can further pinpoint this bug.
>
> ** Changed in: backintime
>       Status: Fix Released => Incomplete
>
> --
> Deleted Files are stored in snapshots forever
> https://bugs.launchpad.net/bugs/406092
> You received this bug notification because you are a direct subscriber
> of the bug.
>

**Comment by Germar** [comment]
(by bratdaking)
OK might not be the problem described in the original bug-report, however I just discovered a way you can store deleted files in the snapshots forever. It might be that I accidentally discovered what this ghost bug is about.

If you include two folders in your snapshots: e.g. /home and /mnt/data make some snapshots and then remove one of the two: e.g. the /mnt/data, BiT does not remove the /mnt/data from the snapshots. It does not update it either, but it stays there as a sort of hardlinked redumentary left-over (in my case even worse as the /mnt/data does not exists even any longer). It is hard to notice though as it also not listed in the Backup Folders list, you really have to walk to the folder (or have a look in the snapshot folders).

The reason is quite logical, rsync only takes the includes into account and compares those to the hardlinked copy of the last snapshot. So any removed files in those are removed from the new snapshot, but because you remove one of the includes: rsync does not longer bother about what is happening inside these folders, and does not take them into account for removal either.
Ok solved, now we need a solution...

Might also be that the .gvfs (gnome virtual file system) file is influencing the delete, this file is completely locked, even for root, and should therefore always be excluded.

**Comment by Germar** [comment]
(by andreas-welcomes-you)
I can confirm the bug, however it only shows up under specific conditions.

- If BiT takes snapshots of my small Temp directory (/home/andreas/Temp) it behaves as expected (meaning files which are deleted before the next snapshot are not anymore visible in the snapshot taken after the deletion of the file.

- BUT if BiT has to backup my complete home directory (/home/andreas) which hosts 30GB, it keeps in the snapshots deleted files. Example: I have a file called foo in ~/Temp. I make a snapshot1. It appears in snapshot1. I delete the file foo. I make a snapshot2. When I browse in BiT in snapshot2, I still see the file foo, althought in "Now" it doesn't show up anymore.

Maybe it depends on the size of the to be backuped directory?

Question:
- How can we trace, what commands are issued by BiT to rsync when making a snapshot? Please let me know, how I can look at these traces (debug output of rsync including command line invocation).

My environment:
Ubuntu 9.04, rsync version 3.0.5  protocol version 30, BiT 0.9.26 installed via "apt-get". Backup is made on a NAS and is mounted as a NFS on my machine.

Note:
Excluding ".gvfs" makes no difference.

**Comment by Germar** [comment]
(by bratdaking)
The exact commands are logged in your messages log (Command "..." returns 0)
Between " " is the command that is issued (it might be better to have a look
in the syslog, as in messages the commands can be truncated.
You can also run it from the command line: backintime -b than all the
messages are also displayed in your terminal.
The return code should be 0, a known issue is the .gvfs that gives an error
code 5888

If you want to adept the rsync command to invoke debug information: open the
/usr/share/backintime/common/snapshot.py file, search for the rsync_prefix:
you can add additional arguments there. (would be good to add an option
debug or something...)

If you could have a thorough look, that would be great, if you find
something keep us posted...

Cheers,
Bart

2009/11/9 Andreas <email address hidden>

> I can confirm the bug, however it only shows up under specific
> conditions.
>
> - If BiT takes snapshots of my small Temp directory (/home/andreas/Temp)
> it behaves as expected (meaning files which are deleted before the next
> snapshot are not anymore visible in the snapshot taken after the
> deletion of the file.
>
> - BUT if BiT has to backup my complete home directory (/home/andreas)
> which hosts 30GB, it keeps in the snapshots deleted files. Example: I
> have a file called foo in ~/Temp. I make a snapshot1. It appears in
> snapshot1. I delete the file foo. I make a snapshot2. When I browse in
> BiT in snapshot2, I still see the file foo, althought in "Now" it
> doesn't show up anymore.
>
> Maybe it depends on the size of the to be backuped directory?
>
> Question:
> - How can we trace, what commands are issued by BiT to rsync when making a
> snapshot? Please let me know, how I can look at these traces (debug output
> of rsync including command line invocation).
>
> My environment:
> Ubuntu 9.04, rsync version 3.0.5  protocol version 30, BiT 0.9.26 installed
> via "apt-get". Backup is made on a NAS and is mounted as a NFS on my
> machine.
>
> Note:
> Excluding ".gvfs" makes no difference.
>
> --
> Deleted Files are stored in snapshots forever
> https://bugs.launchpad.net/bugs/406092
> You received this bug notification because you are a direct subscriber
> of the bug.
>

**Comment by Germar** [comment]
(by sremington)
Excluding ".gvfs" actually resolved the issue for me.

Before BiT I had rolled my own similar back up routine using hard links and rsync and had run into this exact same behavior. What I had discovered then was that rsync was bombing out on .gvfs (permission errors) before it finished running and before it finished deleting files (I was using --delete-during).

Perhaps --exclude=.gvfs should be a default exclude (and not editable by the user).

**Comment by Germar** [comment]
(by bratdaking)
Hey this bug is turning into two:

- the .gvfs problem -> lets continue that one in the default excluded folders bug
                 https://bugs.launchpad.net/backintime/+bug/422132
- the problem of removing an included folder what leads to rudimentary ghost folders in the snapshots
                 -> that one will be continued here

ok?

**Comment by Germar** [comment]
(by andreas-welcomes-you)
After having tried to analyse the bug an several backups (unfortunately they took 2-3h each), I ended up that the bug completely disappeared. Please don't ask me why...

I uninstalled backintime, deleted all configs, deleted the backups from the NFS drive, reinstalled backintime, didn't put ".gvfs" in the exclude list and now everything works fine. No ghost files/folders etc.

Sorry that this bug can't be traced... I can send you the logs I have, but since the bug didn't appear anymore they will not help you. I have no idea how to proceed... Any ideas?

**Comment by Germar** [comment]
(by bratdaking)
Well these are the best bugs off course: ghost bugs that just fix themselves
But this bug is still valid as I found a way to create ghost files in your
snapshots as I described above...

However if your problem reoccurs, please notify us again!

2009/11/12 Andreas <email address hidden>

> After having tried to analyse the bug an several backups (unfortunately
> they took 2-3h each), I ended up that the bug completely disappeared.
> Please don't ask me why...
>
> I uninstalled backintime, deleted all configs, deleted the backups from
> the NFS drive, reinstalled backintime, didn't put ".gvfs" in the exclude
> list and now everything works fine. No ghost files/folders etc.
>
> Sorry that this bug can't be traced... I can send you the logs I have,
> but since the bug didn't appear anymore they will not help you. I have
> no idea how to proceed... Any ideas?
>
> --
> Deleted Files are stored in snapshots forever
> https://bugs.launchpad.net/bugs/406092
> You received this bug notification because you are a direct subscriber
> of the bug.
>

**Comment by Germar** [comment]
(by danleweb)
I confirm, the bug is valid. Your procedure prove it.
rsync has the option " --delete-excluded" to remove excluded files. This can fix the bug but it can disable schedule per include directory (or it need to be redesigned).

**Comment by Germar** [comment]
(by bratdaking)
I know it could be solved by adding the --delete-excluded, however than we
need to redesign the schedule per included folder. It should then hardlink
to snapshots from the same schedule. On hardlinking capable partitions this
issue does not consume more space and it does not take really longer to
accomplish, so he issue is not that important to fix immediately. However we
need to come up with a solution sooner or later...
Maybe we should verify whether the folders in the snapshots all belong to
the included folders (of all schedules)...

Cheers,
Bart

2009/11/13 Dan <email address hidden>

> ** Changed in: backintime
>   Importance: Undecided => High
>
> --
> Deleted Files are stored in snapshots forever
> https://bugs.launchpad.net/bugs/406092
> You received this bug notification because you are a direct subscriber
> of the bug.
>

**Comment by Germar** [comment]
(by bratdaking)
Since we removed the Schedule per Included folder option, I added the --delete-excluded again. This should solve the ghost folders, however I need to verify that still

**Comment by Germar** [comment]
(by esblack)
I've been following along with this bug since it's been a little bit of a headache and just found that if I used the solution posted above (i.e. exclude *.gvfs) and run it from "Back in Time (root)" it works as expected.  The "foo" file vanishes.  If I use the same settings and just use "Back in Time" as user then the "foo" file stays and does not delete from the new snapshot.

Makes me curious how people are running the application.  Anyway I love BiT and now that I have a solid work around it's even better.

Thanks,
Shawn

**Comment by Germar** [comment]
(by danleweb)
Just test it and it seems to work. So no longer schedule per included directory. Use profiles instead.

**Comment by Germar** [comment]
(by philipp-foerster)
just tried again with the newest version 0.9.99beta10 on kde4. files still don't get deleted :( in newer snapshots.
ghost files persist

**Comment by Germar** [comment]
(by danleweb)
Strange, I just tested and it works for me.
If you have the time please do this:
1. create a 'foo' file in one of your included directories
2. take a snapshot (and check that the new file is in the snapshot)
3. remove 'foo' file
4. run 'backintime -b" from command line (and check the new snapshot still contains 'foo')
5. in the output look for a line like: INFO: Command "rsync -aEHAX --delete-excluded  -i --dry-run 
6. run the command (with all parameters) from command lines and attach the output.

**Comment by Germar** [comment]
(by cliff6056)
Experiencing the same as Fifoxtasy also using beta10 on kde4 and new config file.  After running backintime -b the line I think that you request from the output gives a Warning instead of Info and looks as follows:

                                            
WARNING: Command "rsync -aEHAX --delete-excluded  -i --dry-run  --chmod=Da+w --delete  --exclude="/media/disk1part5" --exclude="/home/cliff/.local/share/backintime" --include="/home/cliff/" --include="/home/" --include="/etc/" --exclude=".gvfs" --exclude=".cache*" --exclude="[Cc]ache*" --exclude=".thumbnails*" --exclude="[Tt]rash*" --exclude="*.backup*" --exclude="*~" --exclude="home/cliff/.VirtualBox*" --include="/home/cliff/**" --include="/etc/**" --exclude="*" / "/media/disk1part5/backintime/siduxbox/cliff/1/20100122-172823-113/backup/"" returns 5888                                                                         
INFO: Create hard-links 

Hope this is of assistance.

**Comment by Germar** [comment]
(by philipp-foerster)
WARNING: Command "rsync -aEHAX --delete-excluded  -i --dry-run  --chmod=Da+w --delete  --exclude="/media/Extern/Backups" --exclude="/home/philipp/.local/share/backintime" --include="/home/philipp/" --include="/home/" --include="/media/Data/Fotos/" --include="/media/Data/" --include="/media/" --include="/media/Data/Dropbox/" --include="/etc/" --include="/media/Data/Personal/" --exclude=".gvfs" --exclude=".cache*" --exclude="[Cc]ache*" --exclude=".thumbnails*" --exclude="*~" --exclude="*.backup*" --exclude=".Trash/" --exclude=".tmp/" --exclude="tmp/" --exclude="Temp/" --exclude="temp/" --exclude="/dev/" --exclude="/sys/" --exclude="/proc/" --exclude=".google/picasa/3.0/drive_c/Documents and Settings/philipp/Local Settings/Application Data/Google/Picasa2/db3/" --exclude="lost+found/" --exclude="/var/" --exclude=".local/share/Trash/" --exclude=".~*" --exclude="*.part" --include="/home/philipp/**" --include="/media/Data/Fotos/**" --include="/media/Data/Dropbox/**" --include="/etc/**" --include="/media/Data/Personal/**" --exclude="*" / "/media/Extern/Backups/backintime/philipp-laptop/philipp/1/20100123-000003-141/backup/"" returns 5888                   
INFO: Create hard-links

if i enter the commands terminal, i don't get an error message (except for some files in /etc), only a very long list of all my files. (if i remove /etc from the include list no more error messages). the files that should be deleted are in a normal folder didn't check for /etc.

**Comment by Germar** [comment]
(by cliff6056)
Today I created a new test profile and backed up just a couple of folders to play with.  I found that this new profile worked fine, deleting files from the new snapshots as I believe is expected.  My main backup,  however, which has snapshots that were created before I did a fresh install of backintime-kde4 and also a clean .config/backintime and new backintime-common still retains files that have been deleted, both before the fresh install and ones that have been created since.  Hopefully this is of assistance.

**Comment by Germar** [comment]
(by philipp-foerster)
i thought i'd mention that for me it happens on a fresh install and a fresh backupfolder.

**Comment by Germar** [comment]
(by danleweb)
You should run the command:

rsync -aEHAX --delete-excluded -i --dry-run --chmod=Da+w --delete --exclude="/media/Extern/Backups" --exclude="/home/philipp/.local/share/backintime" --include="/home/philipp/" --include="/home/" --include="/media/Data/Fotos/" --include="/media/Data/" --include="/media/" --include="/media/Data/Dropbox/" --include="/etc/" --include="/media/Data/Personal/" --exclude=".gvfs" --exclude=".cache*" --exclude="[Cc]ache*" --exclude=".thumbnails*" --exclude="*~" --exclude="*.backup*" --exclude=".Trash/" --exclude=".tmp/" --exclude="tmp/" --exclude="Temp/" --exclude="temp/" --exclude="/dev/" --exclude="/sys/" --exclude="/proc/" --exclude=".google/picasa/3.0/drive_c/Documents and Settings/philipp/Local Settings/Application Data/Google/Picasa2/db3/" --exclude="lost+found/" --exclude="/var/" --exclude=".local/share/Trash/" --exclude=".~*" --exclude="*.part" --include="/home/philipp/**" --include="/media/Data/Fotos/**" --include="/media/Data/Dropbox/**" --include="/etc/**" --include="/media/Data/Personal/**" --exclude="*" / "/media/Extern/Backups/backintime/philipp-laptop/philipp/1/20100123-000003-141/backup/"

and see if there is a line for the file that is not removed. If there is one please attach it.

**Comment by Germar** [comment]
(by cliff6056)
Hi Dan.  The experiences on my system (which I reloaded this morning to make sure everything was okay) are:

cliff@siduxbox:~$ rsync -aEHAX --delete-excluded  -v  --chmod=Da+w --delete  --exclude="/media/disk1part4" --exclude="/home/cliff/.local/share/backintime" --include="/home/cliff/" --include="/home/" --include="/etc/" --exclude=".gvfs" --exclude=".cache*" --exclude="[Cc]ache*" --exclude=".thumbnails*" --exclude="[Tt]rash*" --exclude="*.backup*" --exclude="*~" --exclude=".VirtualBox*" --include="/home/cliff/**" --include="/etc/**" --exclude="*" / "/media/disk1part4/backintime/siduxbox/cliff/1/new_snapshot/backup/"
sending incremental file list
rsync: writefd_unbuffered failed to write 4 bytes to socket [sender]: Broken pipe (32)
rsync: mkdir "/media/disk1part4/backintime/siduxbox/cliff/1/new_snapshot/backup" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(595) [Receiver=3.0.7]
rsync: connection unexpectedly closed (9 bytes received so far) [sender]
rsync error: error in rsync protocol data stream (code 12) at io.c(601) [sender=3.0.7]
cliff@siduxbox:~$



As far as I can tell, other than not deleting the deleted file from the backup, it all seems to have worked.  The new snapshot has appeared in the backup folder, even though it says backup failed.

**Comment by Germar** [comment]
(by philipp-foerster)
the output of that command does NOT contain any lines for the files i deleted.

is there anything else i can do to help you?

**Comment by Germar** [comment]
(by philipp-foerster)
still present in beta11. didn't recreate everything though, just updated.

**Comment by Germar** [comment]
(by philipp-foerster)
still present in beta15

**Comment by Germar** [comment]
(by cliff6056)
I found that when I was backing up both /home/me and /etc in the main profile, that the files were not being deleted from my backups.  I have since then created another profile for the /etc, and have /home/me by itself in the "Main Profile".  Since doing this files are deleted, at least from the /home/me which I have checked.  Perhaps this might be of assistance to you.

**Comment by Germar** [comment]
(by danleweb)
Please check user/group/rights:
1. of the original file (before delete)
2. of the file in the correct snapshot
3. of the file in the snapshot where it shouldn't be

**Comment by Germar** [comment]
(by philipp-foerster)
all snapshots have read permissions only, no write permission. in the original folder (/home/...) i have permission to write

**Comment by Germar** [comment]
(by danleweb)
Hi cliff6056,

Just to be sure that I have it right:

1. Old mode: one profile with both /home/me and /etc.
Some files are not deleted from new snapshots even they don't exists anymore.
The files with problems are located in /home/me or /etc or both ?

2. New mode: one profile with /home/me and another with /etc
Looks ok for /home/me.
Please do a test for /etc in the second profile too.

**Comment by Germar** [comment]
(by danleweb)
I think I found the problem. In some cases rsync with "--delete" option does see some files, but with "--delete-before" it does.

Please try beta16 version from testing repository.

**Comment by Germar** [comment]
(by cliff6056)
Using one profile and several backup folders:
Just to report before I go to beta 16 as I just finished testing, I
found that (using a smaller backup /home/me/Documents/FOO and /etc, it
made no difference which order they were in.  The home folder always
retained the FOO, but the etc/BOO file would always delete the BOO.  I
then tested two files from the home folder, the FOO as above, and
/home/me/Downloads/ZOO.  In this case, both the FOO and ZOO would be
deleted as expected.  When I added the /etc/BOO to the snapshot, then
only the BOO was deleted, leaving both the FOO  and ZOO files in the
home folder. 

The etc alone in its own profile deletes as expected.

I will upgrade to beta 16 and let you know.

Dan wrote:
> Hi cliff6056,
>
> Just to be sure that I have it right:
>
> 1. Old mode: one profile with both /home/me and /etc.
> Some files are not deleted from new snapshots even they don't exists anymore.
> The files with problems are located in /home/me or /etc or both ?
>
> 2. New mode: one profile with /home/me and another with /etc
> Looks ok for /home/me.
> Please do a test for /etc in the second profile too.
>
>

**Comment by Germar** [comment]
(by cliff6056)
Tried the same test with 3 folders in one profile, with one folder being
the /etc, and this time (checked twice) nothing was deleted from any. 
Unless I made a mistake in my previous test with 15, I thought the /etc
would at least delete, but definitely not now.

Once I removed the /etc from the include in the test profile, the
remaining files deleted in the /home also deleted from the snapshot as
expected.

Could this be a permissions problem that I have by including both a root
and user directory in one profile, but when separated into different
profiles, work properly? 



Dan wrote:
> I think I found the problem. In some cases rsync with "--delete" option
> does see some files, but with "--delete-before" it does.
>
> Please try beta16 version from testing repository.
>
>

**Comment by Germar** [comment]
(by danleweb)
It looks like /etc to be a problem. On my side the old version (before beta16) didn't work, but now it works with both files from /home/user and from /etc.

I have rsync: 3.0.6

What I did notice is that as a regular user rsync can't explore all items in /etc.

**Comment by Germar** [comment]
(by cliff6056)
Rsync version here is 3.0.7-2.

**Comment by Germar** [comment]
(by danleweb)
Hi cliff6056,

How did you get 3.0.7 ?
What distribution are you using ?

**Comment by Germar** [comment]
(by cliff6056)
Latest Sidux which is based on Debian Sid and KDE 4.3.4.  When I load up
synaptic and check rsync this is the version number that shows. 

Dan wrote:
> Hi cliff6056,
>
> How did you get 3.0.7 ?
> What distribution are you using ?
>
>

**Comment by Germar** [comment]
(by danleweb)
I did some tests with Sidux. Indeed the latest update is 3.0.7.
It seems that the problem is related to rsync.
On my system with 3.0.6 the bug is there but using "--delete-before" as workaround works for me. On the other hand --delete-before is slower and use more memory because it does not use incremental mode. So I prefer to stick with "--delete" option.

From rsync man page:
"
--delete
              This  tells  rsync  to delete extraneous files from the receiving side (ones that aren't on the sending
              side), but only for the directories that are being synchronized.  You must have asked rsync to send the
              whole  directory  (e.g.  “dir”  or  “dir/”) without using a wildcard for the directory's contents (e.g.
              “dir/*”) since the wildcard is expanded by the shell and rsync thus gets a request to transfer individ‐
              ual  files,  not  the  files'  parent  directory.   Files  that are excluded from the transfer are also
              excluded from being deleted unless you use the --delete-excluded option  or  mark  the  rules  as  only
              matching on the sending side (see the include/exclude modifiers in the FILTER RULES section).

              Prior  to rsync 2.6.7, this option would have no effect unless --recursive was enabled.  Beginning with
              2.6.7, deletions will also occur when --dirs (-d) is enabled, but only for directories  whose  contents
              are being copied.

              This  option can be dangerous if used incorrectly!  It is a very good idea to first try a run using the
              --dry-run option (-n) to see what files are going to be deleted.

              If the sending side detects any I/O errors, then the deletion of any files at the destination  will  be
              automatically  disabled.  This  is to prevent temporary filesystem failures (such as NFS errors) on the
              sending side causing a massive deletion of files on the destination.  You can override  this  with  the
              --ignore-errors option.

              The  --delete option may be combined with one of the --delete-WHEN options without conflict, as well as
              --delete-excluded.  However, if none of the --delete-WHEN options are specified, rsync will choose  the
              --delete-during  algorithm when talking to rsync 3.0.0 or newer, and the --delete-before algorithm when
              talking to an older rsync.  See also --delete-delay and --delete-after.
"

The problem appears only if I include /etc as a regular user and I don't have the right to explore everything. This may happen to all folders I don't have enough rights.

If I run BIT as root if works as expected.

**Comment by Germar** [comment]
(by danleweb)
Since the problem came from rsync I'll pass it to won't fix.

**Comment by Germar** [comment]
(by cliff6056)
Many thanks Dan.  BIT is great.  At least the workaround of keeping the / and /home in different profiles works for me, hopefully for others.

**Comment by Germar** [comment]
(by philipp-foerster)
work around works for me as well
maybe this should be in the help

**Comment by Germar** [comment]
(by snearch)
I use backintime 1.0.8 (root) from the PPA on lucid 10.04/3,
and if I rename a file I see two files (the old and the 
new name) in the next Backup.
